crates/biome_service/src/file_handlers/svelte.rs

```diff
@@ -116,21 +132,35 @@ impl ExtensionHandler for SvelteFileHandler {
 
 fn parse(
     _rome_path: &BiomePath,
-    _file_source: DocumentFileSource,
+    file_source: DocumentFileSource,
     text: &str,
     _settings: &Settings,
     cache: &mut NodeCache,
 ) -> ParseResult {
-    let script = SvelteFileHandler::input(text);
-    let file_source = SvelteFileHandler::file_source(text);
-
-    debug!("Parsing file with language {:?}", file_source);
-
-    let parse = parse_js_with_cache(script, file_source, JsParserOptions::default(), cache);
-
-    ParseResult {
-        any_parse: parse.into(),
-        language: Some(file_source.into()),
+    // Use HTML parsing to extract script content with proper context
+    let scripts = SvelteFileHandler::extract_scripts(text, cache);
+    
+    // If we found scripts, use the first one; otherwise fall back to regex approach
+    if let Some(first_script) = scripts.first() {
```

Comment:
This is wrong, we want to process all the script blocks.

Reply:
ah! thats fascinating. i've never used a script module before. 🤔

![Screenshot from 2025-07-06 01-35-21](https://github.com/user-attachments/assets/cfca77ed-28d3-4e6a-862c-d427517bdfae)